pkg/planner/cardinality/selectivity.go

```diff
@@ -119,12 +119,16 @@ func Selectivity(
 
 	extractedCols := make([]*expression.Column, 0, len(coll.Columns))
 	extractedCols = expression.ExtractColumnsFromExpressions(extractedCols, remainedExprs, nil)
-	colIDs := maps.Keys(coll.Columns)
-	slices.Sort(colIDs)
-	for _, id := range colIDs {
-		colStats := coll.Columns[id]
-		col := expression.ColInfo2Col(extractedCols, colStats.Info)
-		if col != nil {
+	slices.SortFunc(extractedCols, func(a *expression.Column, b *expression.Column) int {
+		return cmp.Compare(a.ID, b.ID)
+	})
+	for i, col := range extractedCols {
+		if i != 0 && extractedCols[i].ID == extractedCols[i-1].ID {
+			continue
+		}
```

Comment:
We can use `slices.CompactFunc` from `golang.org/x/exp/slices` to dedup `extractedCols`. I think that will be easier to understand.